app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt

```diff
@@ -36,8 +36,12 @@ class ProfileModelView(private val repository: ProfileRepository) : ViewModel()
   val profile: StateFlow<Profile> = profile_.asStateFlow()
 
   /** The initialisation function of the profile model view. It fetches the profile of the user */
-  suspend fun initAfterLogin() {
-    repository.initAfterLogin()
+  suspend fun initAfterLogin(onSuccess: () -> Unit) {
+    repository.initAfterLogin {
+        profile_.value = it
+        CurrentProfile.currentProfile = it
```

Comment:
CurrentProfile doesn't need to use anymore. You need to use ProfileViewModel instead !